source/dub/platform.d

```diff
@@ -8,43 +8,18 @@
 	It also contains means to match build platforms against a platform
 	specification string as used in package reciptes.
 
-	Copyright: © 2012-2016 rejectedsoftware e.K.
+	Copyright: © 2012-2017 rejectedsoftware e.K.
 	License: Subject to the terms of the MIT license, as written in the included LICENSE.txt file.
 	Authors: Sönke Ludwig
 */
 module dub.platform;
 
 import std.array;
 
-
-/** Determines the full build platform used for the current build.
-
-	Note that the `BuildPlatform.compilerBinary` field will be left empty.
-
-	See_Also: `determinePlatform`, `determineArchitecture`, `determineCompiler`
-*/
-BuildPlatform determineBuildPlatform()
-{
-	BuildPlatform ret;
-	ret.platform = determinePlatform();
-	ret.architecture = determineArchitecture();
-	ret.compiler = determineCompiler();
-	ret.frontendVersion = __VERSION__;
-	return ret;
-}
-
-
-/** Returns a list of platform identifiers that apply to the current
-	build.
-
-	Example results are `["windows"]` or `["posix", "osx"]`. The identifiers
-	correspond to the compiler defined version constants built into the
-	language, except that they are converted to lower case.
-
-	See_Also: `determineBuildPlatform`
-*/
-string[] determinePlatform()
-{
+// archCheck, compilerCheck, and platformCheck are used below and in
+// generatePlatformProbeFile, so they've been extracted into these strings
+// that can be reused.
+enum string platformCheck = q{
```

Comment:
Can you put a `/// private` comment right in from of this and the other `...Check` enums? That will keep them from being included in the documentation. It would be nicer to make them `package(dub.compilers)`, but I'm pretty sure that we still support frontend versions that don't support that syntax.

Reply:
Actually, just let me just do that from the GitHub UI quickly...